Content.Server/TwoHanded/TwoHandedSystem.cs

```diff
@@ -0,0 +1,72 @@
+using Content.Server.Hands.Components;
+using Content.Server.Popups;
+using Content.Shared.Item;
+using Content.Shared.Hands;
+using Content.Shared.Hands.EntitySystems;
+using Robust.Shared.Player;
+
+namespace Content.Server.TwoHanded;
+    internal sealed class TwoHandedSystem : EntitySystem
+    {
+        [Dependency] private readonly HandVirtualItemSystem _virtualItemSystem = default!;
+        [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<TwoHandedComponent,GettingPickedUpAttemptEvent>(PickupAttempt);
+            SubscribeLocalEvent<TwoHandedComponent,GotEquippedHandEvent>(OnEquipped);
+            SubscribeLocalEvent<TwoHandedComponent,GotUnequippedHandEvent>(OnDeEquipped);
+            SubscribeLocalEvent<TwoHandedComponent,VirtualItemDeletedEvent>(OnVirtualItemDeleted);
+        }
+
+        private void PickupAttempt(EntityUid uid, TwoHandedComponent comp, GettingPickedUpAttemptEvent args)
+        {
+            if (args.User == null)
+                return;
+            if (!TryComp<HandsComponent>(args.User, out var hands) || hands is null)
+                return;
+            foreach (var hand in hands.Hands.Values)
+            {
+                if (hand.IsEmpty == true)
+                    continue;
+                args.Cancel();
+                _popupSystem.PopupEntity(Loc.GetString("item-to-heavy"), uid, Filter.Entities(args.User));
+            }
+        }
+        private void OnEquipped(EntityUid uid, TwoHandedComponent comp, GotEquippedHandEvent args)
+        {
+            if (args.User == null)
+                return;
+            if (!TryComp<HandsComponent>(args.User, out var hands) || hands is null)
+                return;            
+            _virtualItemSystem.TrySpawnVirtualItemInHand(args.Equipped, args.User);
+        }
+
+        private void OnDeEquipped(EntityUid uid, TwoHandedComponent comp, GotUnequippedHandEvent args)
+        {
+            if (args.User == null)
+                return;
+            _virtualItemSystem.DeleteInHandsMatching(args.User, args.Unequipped);
+        }
+
+        private void OnVirtualItemDeleted(EntityUid uid, TwoHandedComponent comp, VirtualItemDeletedEvent args)
+        {
+            if (args.User == null)
+                return;
+            if (!TryComp<HandsComponent>(args.User, out var hands) || hands is null)
+                return;
+            if (args.BlockingEntity == uid)
+            {
+                foreach(var hand in hands.Hands)
+                {
+                    if (hand.Value != hands.ActiveHand!)
+                    {
+                        _handsSystem.TryDrop(args.User, hand.Value, null, checkActionBlocker: true);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Isn't this just wielding?

Reply:
Tbf wielding is optional and is a buff bc you can't be disarmed while wielding, this is more of a trade off of using a powerful item (e.g can't, the flags effects wont take place if its not being held, but at the same time you can't use any items/weapons while holding it)

Reply:
Wielding is an optional buff (bc from what I've read you can't be disarmed if wielding a weapon) while the twohanded system is a trade off for using a powerful item (double damage is powerful I believe), e.g you can't use the flags power if your not holding it, and you can't hold it and other items (such as weapons) at the same time

Reply:
I don't see why wieldable can't just handle this, if you really want it disarmable add a bool for it.